modules/nf-core/fqtk/main.nf

```diff
@@ -0,0 +1,49 @@
+process FQTK {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda "bioconda::fqtk=0.2.1"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/fqtk:0.2.1--h9f5acd7_0' :
+        'quay.io/biocontainers/fqtk:0.2.1--h9f5acd7_0' }"
+
+    input:
+    tuple val(meta), path(sample_sheet), val(fastq_readstructure_pairs)
+    // fastq_readstructure_pairs example:
+    // [[<fastq name: string>, <read structure: string>, <path to fastqs: path>], [example_R1.fastq.gz, 150T, ./work/98/30bc..78y/fastqs/]]
+
+    output:
+    // Demultiplexed file name changes depending on the arg '--output-types'
+    tuple val(meta), path('output/*.fq.gz')                         , emit: sample_fastq
+    tuple val(meta), path('output/demux-metrics.txt')               , emit: metrics
+    tuple val(meta), path('output/unmatched*.fq.gz')                , emit: most_frequent_unmatched
+    path "versions.yml"                                             , emit: versions
+
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    // Join the absolute path from UNTAR.out.untar to the fastq file names
```

Comment:
Does this work if the reads haven't been run through UNTAR?

Reply:
Cool, just a possible foot gun for users.

Reply:
You are not wrong, I can open an issue and work on an option for users to not have to provide tared reads at a later point. Sound ok?